torch/nn/parallel/distributed.py

```diff
@@ -453,6 +459,9 @@ def __init__(self, module, device_ids=None,
         # Whether to perform input tensor CPU to GPU copies on a side-stream
         self.use_side_stream_for_tensor_copies = os.environ.get("PYTORCH_DDP_USE_SIDE_STREAM", "1") == "1"
 
+        # CPU-only process group used for debug synchronization in DDP when
+        # debug mode is enabled.
+        self._cpu_pg = self._create_gloo_pg()
```

Comment:
could we move this line to after line 476?

Reply:
This would be ideal, but it's challenging since some non-zero ranks may crash/hang at `dist._verify_model_across_ranks(self.process_group, parameters)` because of model mismatch. To successfully create the pg we need all ranks to call into it and having it after 476 could cause issues if some ranks crash early.